### PR TITLE
Add pull to refresh

### PR DESCRIPTION
**Pull request by karllindmark**
NOTE: THIS IS **NOT** DONE YET, JUST CREATED TO SHARE PROGRESS.

@peter-budo

Now every screen supports "swipe to refresh", which is nice. However, we'll need to do some polishing on the UI side of it ("empty screens" ++). :8ball:

### CODE REVIEW

**Comment by peter-budo** [comment]
Did you checked https://github.com/google/iosched before you started this?

**Comment by karllindmark** [comment]
They have a similar setup in
https://github.com/google/iosched/blob/master/android/src/main/java/com/google/samples/apps/iosched/ui/BaseActivity.java
and
https://github.com/google/iosched/blob/master/android/src/main/res/layout/activity_browse_sessions.xml
(using custom MultiSwipeRefreshLayout).  ☺

**Comment by peter-budo** [comment]
Cool just checking that we are not re-inventing the wheel

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/base/ui/BaseLoadingFragment.java [hunk 1]
```diff
@@ -27,7 +30,9 @@
 
 import java.util.Arrays;
 
-public abstract class BaseLoadingFragment extends BaseFragment implements Response.ErrorListener {
+public abstract class BaseLoadingFragment
```
Comment (peter-budo) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/base/ui/BaseLoadingFragment.java:33]: What setup you are using that these class declarations are being moved to number of lines?

Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/base/ui/BaseLoadingFragment.java:33]: That's a good question, not really sure what changed that. :8ball:

**Comment by karllindmark** [comment]
This has been reworked in #156.

**Pull request closed without merging**